### PR TITLE
TST: Disable flaky SAMP server tests in CI and centralize test env vars

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to:

* Disable flaky SAMP server tests in CI because they fail a lot with `Timeout expired!` errors nowadays. Example log: https://github.com/astropy/astropy/actions/runs/6502698863/job/17662125118
* Move `CI` and `IS_CRON` to tests helper module so they can be shared across the test suite.

- [x] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

**Comment by pllim** [comment]
pre-commit.ci autofix

**Comment by pllim** [comment]
Weird, why do I see a totally unrelated ResourceWarning from #15281 ?

```
 ResourceWarning: unclosed file <_io.BufferedReader name='C:\\\\...\\test_votable_parquet.vot.parquet'>
```

**Comment by bsipocz** [comment]
@pllim - lovely, `ResourceWarning`s in windows are my favorite. Thanks for finding a workaround for it though, my setup is not exactly suitable for debugging windows specific issues.

File: astropy/samp/tests/test_standard_profile.py [hunk 1]
```diff
@@ -13,6 +14,7 @@ def setup_module(module):
     conf.use_internet = False
 
 
+@pytest.mark.skipif(CI, reason="flaky in CI")
```
Comment (bsipocz) [inline astropy/samp/tests/test_standard_profile.py:17]: have you thought about using pytest rerunfailures instead? I had a reasonably good experience with it in astroquery/pyvo for flaky tests that would usually pass after a CI rerun.

**Comment by pllim** [comment]
pre-commit.ci autofix

File: astropy/samp/tests/test_standard_profile.py (hunk 1 above)
Comment (pllim) [reply astropy/samp/tests/test_standard_profile.py:17]: Not sure if re-running this adds any benefit as a whole. SAMP code is pretty stable by now and slated to be moved over to pyvo at some point. But I'll keep that in mind for other stuff in the future. Thanks!

**Comment by pllim** [comment]
Now I get this failure in devdeps, but does not seem related to anything I am trying to fix here, @mhvk ?

```
astropy/units/tests/test_quantity_erfa_ufuncs.py::TestPVUfuncs::test_pvstar - AssertionError: 
Arrays are not equal

Mismatched elements: 2 / 2 (100%)
Max absolute difference: 0.00078125
Max relative difference: inf
 x: array([0.000781, 0.000781])
 y: array([0., 0.])
```

File: astropy/io/votable/tree.py [hunk 2]
```diff
@@ -3132,32 +3132,38 @@ def _parse_parquet(self, iterator, config):
                 NotImplementedError,
             )
 
-        # Hack to keep windows working
         try:
-            fd = urllib.request.urlopen(href)
-        except urllib.error.URLError:
             # Hack to keep windows working
```
Comment (mhvk) [inline astropy/io/votable/tree.py:3136]: Just one time is probably enough - remove if you still need to iterate (otherwise, don't bother!)

**Review by mhvk — APPROVED** [review]
I like the factoring out of `CI` and `IS_CRON`! One comment that can safely be ignored...

File: astropy/io/votable/tree.py (hunk 2 above)
Comment (pllim) [reply astropy/io/votable/tree.py:3136]: You mean flatten the nested `try`? I am not sure if that is doable given the `if` within the outer `try`. The goal of the outer try is to catch any flavor of `fd` and then attempt to close it after Table parsing. Suggestion is welcome!

**Comment by pllim** [comment]
Looks like the codecov upload got stuck in processing. I think we can ignore the coverage "failure".

File: astropy/io/votable/tree.py (hunk 2 above)
Comment (mhvk) [reply astropy/io/votable/tree.py:3136]: No, I just meant to remove the comment, since it is done another time. But let's ignore it!

**Comment by mhvk** [comment]
Agreed, let's merge!

File: astropy/io/votable/tree.py (hunk 2 above)
Comment (bsipocz) [reply astropy/io/votable/tree.py:3136]: Keep the comment (duplicating was a squash leftover, but having it once is useful) as the whole except part was needed for windows only.

**Pull request merged**